core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts

```diff
@@ -170,20 +170,27 @@ export class DotBlockEditorComponent implements OnInit, OnDestroy {
         this.editor.view.dispatch(tr);
     }
 
-    /**
-     * This methods get the customBlocks variable to retrieve the custom modules as Objects.
-     * Validates that there is customBlocks defined.
-     */
-
-    private async getCustomBlocks(): Promise<AnyExtension[]> {
-        let data: CustomBlock;
+    private getParsedCustomBlocks(): RemoteCustomExtentions {
         try {
-            data = JSON.parse(this.customBlocks);
+            return JSON.parse(this.customBlocks);
         } catch (e) {
+            // eslint-disable-next-line no-console
+            console.log('Parsed customBlocks error:', e);
             console.warn('JSON parse fails, please check the JSON format.');
 
-            return [];
+            return {
+                extensions: []
+            };
         }
+    }
+
+    /**
+     * This methods get the customBlocks variable to retrieve the custom modules as Objects.
+     * Validates that there is customBlocks defined.
+     */
+
```

Comment:
Use this extension which provides a more structured doc: https://marketplace.visualstudio.com/items?itemName=oouo-diogo-perdigao.docthis